aws-core/src/main/java/com/amplifyframework/auth/CognitoCredentialsProvider.kt

```diff
@@ -26,6 +26,7 @@ import kotlin.coroutines.suspendCoroutine
 /**
  * Wrapper to provide credentials from Auth synchronously and asynchronously
  */
+@Deprecated("This interface was released with public visibility, but is not intended to be consumed.")
 open class CognitoCredentialsProvider : AuthCredentialsProvider {
```

Comment:
I'm having second thoughts about this. If the customer uses the Kotlin SDK directly, this CredentialsProvider allows them to easily use a pre-built one that fetches credentials via `Amplify.Auth`.

Reply:
I removed the deprecation. This gives customers the AWSCredentialsProvider that we intend for them to implement on their own, while keeping this as a prebuilt one that can be passed into Kotlin SDK.